adapters/repos/db/vector/hnsw/deserializer.go

```diff
@@ -153,8 +153,13 @@ func (c *Deserializer) ReadNode(r io.Reader, res *DeserializationResult) error {
 	}
 
 	if res.Nodes[id] == nil {
-		res.Nodes[id] = &vertex{level: int(level), id: id, connections: make(map[int][]uint64)}
+		res.Nodes[id] = &vertex{level: int(level), id: id, connections: make([][]uint64, level+1)}
 	} else {
+		if int(level) >= len(res.Nodes[id].connections) {
+			newConns := make([][]uint64, level+1)
+			copy(newConns, res.Nodes[id].connections)
```

Comment:
Copy returns the number of copied bytes, maybe check the return value?

Reply:
Can you elaborate? Check them and do what with the info?

Reply:
Just checked the docu again, on second thought I think it is unnecessary